python/hidet/ir/primitives/cpu/avx.py

```diff
@@ -69,6 +87,51 @@ def register_primitive_functions():
     for name, codegen_name, func_type in functions:
         register_primitive_function(name=name, func_or_type=func_type, codegen_name=codegen_name)
 
+    from hidet.lang import script, attrs
+    from hidet.ir.dtypes import f32x8, f32
+    from hidet.ir.func import Function
+
+    @script
+    def avx_x86_f32x8_sum(x: f32x8) -> f32:
+        attrs.func_kind = "cpu_internal"
+        attrs.func_name = "avx_x86_float32x8_sum"
+        sum_vec = call_primitive_func(
+            'avx_x86_float32x4_add',
+            [
+                call_primitive_func('avx_x86_float32x8_extract_half', [x, 0b0]),
+                call_primitive_func('avx_x86_float32x8_extract_half', [x, 0b1]),
+            ],
+        )
+        sum_vec = call_primitive_func('avx_x86_float32x4_hadd', [sum_vec, sum_vec])
+        sum_vec = call_primitive_func('avx_x86_float32x4_hadd', [sum_vec, sum_vec])
+        return call_primitive_func('avx_x86_float32x4_extract_last', [sum_vec])
+
+    assert isinstance(avx_x86_f32x8_sum, Function)
+    register_primitive_function(avx_x86_f32x8_sum.name, avx_x86_f32x8_sum)
+
+    @script
+    def avx_x86_f32x8_scalar_max(x: f32x8) -> f32:
+        attrs.func_kind = "cpu_internal"
+        attrs.func_name = "avx_x86_float32x8_scalar_max"
+        y = call_primitive_func('avx_x86_float32x8_permute_2f128', [x, x, 1])
+        m1 = call_primitive_func('avx_x86_float32x8_max', [x, y])
+        m2 = call_primitive_func('avx_x86_float32x8_permute', [m1, 0b01001110])
+        m3 = call_primitive_func('avx_x86_float32x8_max', [m1, m2])
+        m4 = call_primitive_func('avx_x86_float32x8_permute', [m3, 0b10110001])
+        m = call_primitive_func('avx_x86_float32x8_max', [m3, m4])
+        return call_primitive_func('avx_x86_float32x8_extract_last', [m])
```

Comment:
I recommand to move these user-defined functions over avx (not the ones provided by underlying vector library) to another file called `avx_helpers.py`.

For functions like `avx_x86_float32x4_extract_last`, we also need to define a wrapper function like
```python
def avx_x86_float32x4_extract_last(x: Expr) -> Call:
    return call_primitive_func('avx_x86_float32x4_extract_last', [x])
```

Reply:
In the new file, we directly use `avx_x86_float32x4_extract_last(...)` in the hidet script, instead of calling `call_primitive_func`.